v1/coordinator/points_writer_internal_test.go

```diff
@@ -12,12 +15,17 @@ func TestSgList_ShardGroupAt(t *testing.T) {
 	}
 
 	list := sgList{
-		{ID: 1, StartTime: day(0), EndTime: day(1)},
-		{ID: 2, StartTime: day(1), EndTime: day(2)},
-		{ID: 3, StartTime: day(2), EndTime: day(3)},
-		// SG day 3 to day 4 missing...
-		{ID: 4, StartTime: day(4), EndTime: day(5)},
-		{ID: 5, StartTime: day(5), EndTime: day(6)},
+		items: meta.ShardGroupInfos{
+			{ID: 1, StartTime: day(0), EndTime: day(1)},
+			{ID: 2, StartTime: day(1), EndTime: day(2)},
+			{ID: 3, StartTime: day(2), EndTime: day(3)},
+			// SG day 3 to day 4 missing...
+			{ID: 4, StartTime: day(4), EndTime: day(5)},
+			{ID: 5, StartTime: day(5), EndTime: day(6)},
+		},
+		needsSort: true,
+		earliest:  day(0),
+		latest:    day(6),
 	}
```

Comment:
It might be better to use the `sgList.Add` method to add each day rather than directly initializing `list`. That way we can be sure the logic to handle sorting and setting `earliest` and `latest` works correctly.